urbackupserver/www2/src/features/status/LastBackups.tsx

```diff
@@ -0,0 +1,115 @@
+import { useEffect } from "react";
+import {
+  Field,
+  ProgressBar,
+  TableCellLayout,
+} from "@fluentui/react-components";
+
+import {
+  ClientProcessActionTypes,
+  ClientProcessItem,
+  StartType,
+  StatusClientItem,
+} from "../../api/urbackupserver";
+import { useBackupResult } from "./BackupResultContext";
+import { formatDatetime } from "./formatDatetime";
+
+export function LastFileBackup(item: StatusClientItem) {
+  const formattedLastBackup =
+    item.lastbackup === 0 ? "Never" : formatDatetime(item.lastbackup);
+
+  const fileBackupProcesses = item.processes.filter(
+    (p) =>
+      [
+        ClientProcessActionTypes.INCR_FILE,
+        ClientProcessActionTypes.FULL_FILE,
+      ].includes(p.action) && p.pcdone > -1,
+  );
+
+  const validBackups = ["incr_file", "full_file"] as StartType[];
+
+  return (
+    <TableCellLayout>
+      <div>{formattedLastBackup}</div>
+      <ProcessResult
+        id={item.id}
+        processes={fileBackupProcesses}
+        validBackups={validBackups}
+      />
+    </TableCellLayout>
+  );
+}
+
+export function LastImageBackup(item: StatusClientItem) {
+  const formattedLastBackup =
+    item.lastbackup_image === 0
+      ? "Never"
+      : formatDatetime(item.lastbackup_image);
+
+  const imageBackupProcesses = item.processes.filter(
+    (p) =>
+      [
+        ClientProcessActionTypes.INCR_IMAGE,
+        ClientProcessActionTypes.FULL_IMAGE,
+      ].includes(p.action) && p.pcdone > -1,
+  );
+
+  const validBackups = ["incr_image", "full_image"] as StartType[];
+
+  return (
+    <TableCellLayout>
+      <div>{formattedLastBackup}</div>
+      <ProcessResult
+        id={item.id}
+        processes={imageBackupProcesses}
+        validBackups={validBackups}
+      />
+    </TableCellLayout>
+  );
+}
+
+function ProcessResult({
+  id,
+  processes,
+  validBackups,
+}: {
+  id: number;
+  processes: ClientProcessItem[];
+  validBackups: StartType[];
+}) {
+  const { getResultById, clearResultById } = useBackupResult();
+
+  const result = getResultById(id);
+
+  if (
+    processes.length === 0 &&
+    result &&
+    validBackups.includes(result.start_type)
+  ) {
+    if (!result.start_ok) {
+      return <div>Starting backup failed</div>;
+    }
+
+    return <div>Queued backup</div>;
+  }
+
+  // Clear backup result to not show outdated status once status table is refetched
+  // after all processes are complete i.e. processes === [].
+  useEffect(() => {
+    if (result) {
+      clearResultById(id, result.start_type);
+    }
+  }, []);
```

Comment:
This looks like it might cause (performance) issues?
Is there some other way to solve this, including something like only displaying the message for a certain amount of time and then removing it again?

Reply:
We can also adjust what the server returns if you have some recommendations...

Reply:
Thank you for pointing this out!

I added a commit which does it in a simpler and more performant way, taking your recommendation:
- keeps a list of backup results
- newest backup results are displayed on the table
- each list of backup results is **cleared every 3s** (from the highest component provider), starting from the oldest backup results, if any

Please let me know if this seems acceptable.